sdk/src/main/java/com/silanis/esl/sdk/examples/AttachmentFileExample.java

```diff
@@ -0,0 +1,93 @@
+import com.silanis.esl.sdk.DocumentType;
+import com.silanis.esl.sdk.Signer;
+import com.silanis.esl.sdk.builder.DocumentBuilder;
+import com.silanis.esl.sdk.builder.SignatureBuilder;
+import com.silanis.esl.sdk.builder.SignerBuilder;
+import com.silanis.esl.sdk.builder.internal.StreamDocumentSource;
+
+import java.io.InputStream;
+import java.util.List;
+
+import static com.silanis.esl.sdk.builder.AttachmentRequirementBuilder.newAttachmentRequirementWithName;
+import static com.silanis.esl.sdk.builder.PackageBuilder.newPackageNamed;
+
+public class AttachmentFileExample extends SDKSample {
+
+    private InputStream attachmentInputStream1;
+    private Signer signer1;
+    public static final String NAME1 = "Driver's license";
+    public static final String DESCRIPTION1 = "Please upload a scanned copy of your driver's license.";
+    public static final String SIGNER1_ID = "signer1Id";
+    private String attachment1Id;
+    private List<AttachmentRequirement> signer1Attachments;
+    private AttachmentRequirement signer1Att1;
+    public static final String ATTACHMENT_FILE_NAME1 = "The attachment1 for signer1.pdf";
+    public List<AttachmentFile> filesAfterUpload;
+    public List<AttachmentFile> filesAfterDelete;
+
+    public static void main(String... args) {
+        new AttachmentFileExample().run();
+    }
+
+    public AttachmentFileExample() {
+        this.attachmentInputStream1 = this.getClass().getClassLoader().getResourceAsStream("document-for-anchor-extraction.pdf");
+    }
+
+    @Override
+    protected void execute() {
+
+        // Signer1 with 1 attachment requirement
+        signer1 = SignerBuilder.newSignerWithEmail(email1)
+                .withFirstName("John")
+                .withLastName("Smith")
+                .withCustomId(SIGNER1_ID)
+                .withAttachmentRequirement(newAttachmentRequirementWithName(NAME1)
+                        .withDescription(DESCRIPTION1)
+                        .isRequiredAttachment()
+                        .build())
+                .build();
+
+
+        DocumentPackage superDuperPackage = newPackageNamed(getPackageName())
+                .describedAs("This is a package created using the eSignLive SDK")
+                .withSigner(signer1)
+                .withDocument(DocumentBuilder.newDocumentWithName("test document")
+                        .fromStream(documentInputStream1, DocumentType.PDF)
+                        .withSignature(SignatureBuilder.signatureFor(email1)
+                                .build())
+                        .build())
+                .build();
+
+        packageId = eslClient.createAndSendPackage(superDuperPackage);
+
+        reloadPackage();
+
+        byte[] attachment1ForSigner1FileContent = new StreamDocumentSource(attachmentInputStream1).content();
+        eslClient.uploadAttachment(packageId, signer1Att1.getId(), ATTACHMENT_FILE_NAME1,
+                attachment1ForSigner1FileContent, SIGNER1_ID);
+
+        reloadPackage();
```

Comment:
You don't need to retrieve everything.
You can change the code as shown below.

```
        retrievedPackage = eslClient.getPackage(packageId);
        signer1Attachments = retrievedPackage.getSigner(email1).getAttachmentRequirements();
        signer1Att1 = signer1Attachments.get(0);
```

Reply:
I've removed reloadPackage method and replaced it as you suggested.